olcTestApp.cpp

```diff
@@ -1,36 +1,67 @@
 #define OLC_PGE_APPLICATION
 #include "external/olcPixelGameEngine/olcPixelGameEngine.h"
 
-class Example : public olc::PixelGameEngine
+class TictacToeWindow : public olc::PixelGameEngine
 {
 public:
-	Example()
-	{
-		sAppName = "Example";
+	TictacToeWindow() {
+		sAppName = "TictacToe";
 	}
 
-public:
-	bool OnUserCreate() override
-	{
-		// Called once at the start, so create things here
-		return true;
+	bool OnUserCreate() override {
+    // Set up the initial state of the game
+    // ...
+
+    // Draw the lines for the 3x3 grid
+	draw_grid_lines();
+
+    return true;
 	}
 
-	bool OnUserUpdate(float fElapsedTime) override
-	{
-		// called once per frame
-		for (int x = 0; x < ScreenWidth(); x++)
-			for (int y = 0; y < ScreenHeight(); y++)
-				Draw(x, y, olc::Pixel(rand() % 255, rand() % 255, rand()% 255));	
+	bool OnUserUpdate(float fElapsedTime) override {
+    // Handle user input and update the game state
+		if(p1_turn) {
+			color = olc::RED;
+		}
+		else {
+			color = olc::BLUE;
+		}
+
+		change_tile_color(color);
 		return true;
 	}
+
+	private:
+	bool p1_turn = true;
+	olc::Pixel color {};
+
+	void draw_grid_lines() {
+		int w = ScreenWidth() / 3;
+    	int h = ScreenHeight() / 3;
+
+    	for (int i = 1; i < 3; i++) {
+        	DrawLine(i * w, 0, i * w, ScreenHeight(), olc::WHITE);
+        	DrawLine(0, i * h, ScreenWidth(), i * h, olc::WHITE);
+    	}
+	}
+
+	void change_tile_color(olc::Pixel color) {
+		if (GetMouse(0).bPressed) {
+			// Calculate which square was clicked
+			int x = GetMouseX() / (ScreenWidth() / 3);
```

Comment:
Nitpick: OLC's API uses std::int32_t = 32 bit always. You use int >= 16 bit, depending on platform.